### PR TITLE
Switch from `dart:html` to `package:web`

### PR DESCRIPTION
**Pull request by GleammerRay**
`dart:html` is not supported by Flutter web WASM (`flutter build web --wasm`) compilation. See more at https://docs.flutter.dev/platform-integration/web/wasm#known-limitations.

### CODE REVIEW

**Comment by GleammerRay** [comment]
`package:web` 0.3.0 is not supported on current stable Flutter version.

**Comment by Fintasys** [comment]
Thanks for your PR 🙏 Please give me some time to check it

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -10,7 +10,7 @@ on:
   workflow_dispatch:
 
 env:
-  flutter_version: "3.16.0"
+  flutter_version: "3.19.3"
```
Comment (Fintasys) [inline .github/workflows/main.yml:13]: Is this required for web 0.51 or is this change out of scope?

File: pubspec.yaml [hunk 2]
```diff
@@ -14,6 +14,7 @@ dependencies:
     sdk: flutter
   plugin_platform_interface: ^2.0.2
   shared_preferences: ^2.0.15
+  web: ^0.5.1
```
Comment (Fintasys) [inline pubspec.yaml:17]: Note: Migration-Guide https://dart.dev/interop/js-interop/package-web

File: .github/workflows/main.yml (hunk 1 above)
Comment (GleammerRay) [reply .github/workflows/main.yml:13]: `package:web 0.3.0` is not supported on Flutter 3.16.0.

Comment (GleammerRay) [reply .github/workflows/main.yml:13]: I set it to the stable cherry picks Flutter version I'm using but feel free to set to any lower stable version that works.

Comment (Fintasys) [reply .github/workflows/main.yml:13]: I think we should then increase the min version inside pubspec as well
Currently it's still `flutter: ">=3.16.0"`

Comment (GleammerRay) [reply .github/workflows/main.yml:13]: Done. It's not important for the Dart tool but it does add some clarity for humans. Well spotted!

**Review by Fintasys — APPROVED** [review]
LGTM ! Thx for doing the migration and enable the package be compatible with WASM 👍

**Pull request merged**